primer/src/Primer/Core/Type.hs

```diff
@@ -44,7 +47,7 @@ data Type' a
   | TFun a (Type' a) (Type' a)
   | TVar a TyVarName
   | TApp a (Type' a) (Type' a)
-  | TForall a TyVarName Kind (Type' a)
+  | TForall a TyVarName (Kind' ()) (Type' a)
```

Comment:
For clarity: we intentionally don't put metadata on the kind here as is out of scope for this PR. See FR #1098. Here we are only interested in type parameters, and not foralls

Reply:
Since I already need to rebase, I'll add this to the commit message when doing so.